### ISSUE TITLE
Using union of type variables for generic parameter causes reportGeneralTypeIssues

### ISSUE THREAD

**Issue by saryou** [microsoft/pyright#1723]
**Describe the bug**

Using union of type variables for generic parameter causes reportGeneralTypeIssues

**To Reproduce**

pyright reports `error: Argument of type "list[int | None]" cannot be assigned to parameter "v" of type "List[T@f | None]" in function "f"` for this code:

```
from typing import TypeVar, List, Optional


T = TypeVar('T')


def f(v: List[Optional[T]]) -> T:
    assert (value := v[0])
    return value


v: List[Optional[int]] = [1, None]
f(v)
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.129, which I just published. It will also be included in the next release of pylance.
